sdk/eventhub/azure-eventhub/azure/eventhub/_producer.py

```diff
@@ -49,6 +55,7 @@ def _set_trace_message(event_datas, parent_span=None):
     # type: (Iterable[EventData], Optional[AbstractSpan]) -> Iterable[EventData]
     for ed in iter(event_datas):
```

Comment:
Is it safe to assume that you can exhaust the iterable passed in?

Reply:
It's implemented as a list all the way through, but I can wrap that in a list if you wish, just future's sake

Reply:
Ok, you almost got me here, _set_trace_message is an iterator itself (see it's using yield).